executor-service/src/main/java/executor/service/facade/parallel/ConfigurableThreadPool.java

```diff
@@ -0,0 +1,53 @@
+package executor.service.facade.parallel;
+
+import executor.service.model.ThreadPoolConfig;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Logger;
+
+public class ConfigurableThreadPool extends ThreadPoolExecutor {
+
+    private final ThreadLocal<Long> startTime = new ThreadLocal<>();
```

Comment:
It can lead to a memory leak in a thread pool where the same threads are used to execute many tasks.
It’s a good practice to remove the ThreadLocal value when you’re done with it (in the `afterExecute` method).